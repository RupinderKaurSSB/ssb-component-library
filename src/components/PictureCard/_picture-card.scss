.ssb-picture-card {
  box-sizing: border-box;
  color: $ssb-white !important;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  height: 400px;
  justify-content: flex-end;
  padding: 64px 20px;
  position: relative;
  width: 280px;
  background-image: linear-gradient(to top, rgba(39, 66, 71, .6) 25%, rgba(39, 66, 71, .3) 50%, rgba(39, 66, 71, 0) 75%);

  .image-background {
    height: 400px;
    left: 0;
    overflow: hidden;
    position: absolute;
    top: 0;
    width: 280px;
    z-index: -1;

    img {
      transform: scale(1);
      transition: transform .25s ease, -webkit-transform .25s ease;
    }

  }

  &:hover {

    .image-background img {
      transform: scale(1.1);
    }

  }

  .il-type {
    @include roboto;
    font-size: 16px;
    line-height: 1.25;
    margin-bottom: 12px;
  }

  .il-title {
    @include roboto-condenced;
    font-size: 28px;
    font-weight: bold;
    line-height: 1.43;
    margin-bottom: 12px;
  }

  &.horizontal {
    height: 212px;
    padding: 40px 20px;
    width: 380px;

    .image-background {
      height: 212px;
      width: 380px;
    }
  }
}
