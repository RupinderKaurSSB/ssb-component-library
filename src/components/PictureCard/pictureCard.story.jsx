import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import PictureCard from './index';
import cardImage from './cards_bilde.jpg';

const sup = (() => {
	console.log('Image link clicked');
});

storiesOf('Picture Card', module).addDecorator(centered)
	.add('Default', () => (
		<PictureCard imageSrc={cardImage} altText="img alt" onClick={sup} type="Type" title="Tittel" />
	))
	.add('Horizontal', () => (
		<PictureCard orientation="horizontal" imageSrc={cardImage} altText="img alt" onClick={sup} type="Type" title="Tittel" />
	));
