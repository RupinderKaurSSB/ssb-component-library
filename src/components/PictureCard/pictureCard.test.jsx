import React from 'react';
import { shallow } from 'enzyme';
import PictureCard from './index';
import cardImage from './cards_bilde.jpg';

describe('ImageLink component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(
			<PictureCard imageSrc={cardImage} onClick={() => {}} type="Type" altText="img alt" title="Tittel" />
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('Sets default orientation className', () => {
		const wrapper = shallow(<PictureCard imageSrc={cardImage} altText="alt img" onClick={() => {}} type="Type" title="Tittel" />);
		expect(wrapper.hasClass('vertical')).toEqual(true);
	});
	test('Switches className on orientation change', () => {
		const wrapper = shallow(<PictureCard orientation="horizontal" imageSrc={cardImage} altText="alt img" onClick={() => {}} type="Type" title="Tittel" />);
		expect(wrapper.hasClass('horizontal')).toEqual(true);
	});
});
