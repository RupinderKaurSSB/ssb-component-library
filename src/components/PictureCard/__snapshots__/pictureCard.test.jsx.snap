// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLink component Matches the snapshot 1`] = `
<a
  className="ssb-picture-card vertical "
  onClick={[Function]}
>
  <div
    className="image-background"
  >
    <img
      alt=""
    />
  </div>
  <div
    className="overlay"
  >
    <span
      className="il-type"
    >
      Type
    </span>
    <span
      className="il-title"
    >
      Tittel
    </span>
    <ArrowRight
      className="arrow-icon"
      color="currentColor"
      size={32}
    />
  </div>
</a>
`;
